remotes/mount.go

```diff
@@ -151,16 +151,25 @@ func (r *remoteReaderAt) ReadAt(p []byte, off int64) (int, error) {
 }
 
 type descriptorAccumulator struct {
-	descriptors []ocischemav1.Descriptor
+	descriptors     []ocischemav1.Descriptor
+	childrenHandler images.HandlerFunc
+	resolver        docker.ResolverBlobMounter
+	targetRepo      string
 }
 
 func (a *descriptorAccumulator) Handle(ctx context.Context, desc ocischemav1.Descriptor) ([]ocischemav1.Descriptor, error) {
+	_, _, err := a.resolver.Resolve(ctx, fmt.Sprintf("%s@%s", a.targetRepo, desc.Digest))
+	if err == nil {
+		fmt.Fprintf(os.Stderr, "Ignoring descriptor %s, as it is already present in target repository\n", desc.Digest)
+		// already exsits, do not process
```

Comment:
typo here ("exsits").

I wonder though -- should we go a step further here and avoid the `Resolve` for images which we already dealt with in this session by keeping a little cache?

Reply:
The problem here, is that we need to process everything in reverse order, because pushing a manifest requires all blobs to be present. So if we want that, we need to deduplicate descriptors after the accumulation phase to make sure we keep the "Latest" seen in the accumulation phase.